mlflow/pytorch/pytorch_autolog.py

```diff
@@ -0,0 +1,197 @@
+            """
+            if hasattr(early_stop_obj, "monitor"):
+                try_mlflow_log(mlflow.log_param, "monitor", early_stop_obj.monitor)
+
+            if hasattr(early_stop_obj, "mode"):
+                try_mlflow_log(mlflow.log_param, "mode", early_stop_obj.mode)
+
+            if hasattr(early_stop_obj, "patience"):
+                try_mlflow_log(mlflow.log_param, "patience", early_stop_obj.patience)
+
+            if hasattr(early_stop_obj, "min_delta"):
+                try_mlflow_log(mlflow.log_param, "min_delta", early_stop_obj.min_delta)
+
+            if hasattr(early_stop_obj, "stopped_epoch"):
+                try_mlflow_log(mlflow.log_param, "stopped_epoch", early_stop_obj.stopped_epoch)
+
+        @staticmethod
+        def _early_stop_check(early_stop_callback):
+            """
+            Logs all early stopping metrics
+
+            :param early_stop_callback: Early stopping callback object
+            """
+            if early_stop_callback.stopped_epoch != 0:
+
+                if hasattr(early_stop_callback, "stopped_epoch"):
+                    try_mlflow_log(
+                        mlflow.log_metric, "Stopped_Epoch", early_stop_callback.stopped_epoch
+                    )
+                    restored_epoch = early_stop_callback.stopped_epoch - max(
+                        1, early_stop_callback.patience
+                    )
+                    try_mlflow_log(mlflow.log_metric, "Restored_Epoch", restored_epoch)
+
+                if hasattr(early_stop_callback, "best_score"):
+                    try_mlflow_log(mlflow.log_metric, "Best_Score", early_stop_callback.best_score)
+
+                if hasattr(early_stop_callback, "wait_count"):
+                    try_mlflow_log(mlflow.log_metric, "Wait_Count", early_stop_callback.wait_count)
+
+    def _run_and_log_function(self, original, args, kwargs):
+        """
+        This method would be called from patched fit method and
+        It adds the custom callback class into callback list.
+        """
+        if not mlflow.active_run():
+            try_mlflow_log(mlflow.start_run)
+            auto_end_run = True
+        else:
+            auto_end_run = False
+
+        if not any(isinstance(callbacks, __MLflowPLCallback) for callbacks in self.callbacks):
+            self.callbacks += [__MLflowPLCallback()]
+        result = original(self, *args, **kwargs)
+        if auto_end_run:
+            try_mlflow_log(mlflow.end_run)
+        return result
+
+    @gorilla.patch(pl.Trainer)
+    def fit(self, *args, **kwargs):
+        """
+        Patching trainer.fit method to add autolog class into callback
+        """
+        original = gorilla.get_original_attribute(pl.Trainer, "fit")
+        return _run_and_log_function(self, original, args, kwargs)
+
+    settings = gorilla.Settings(allow_hit=True, store_hit=True)
+    gorilla.apply(gorilla.Patch(pl.Trainer, "fit", fit, settings=settings))
```

Comment:
```suggestion
    wrap_patch(pl.Trainer, "fit", fit)
```
Can you use the `wrap_patch` function here to prevesr the original arributes (e.g. `__doc__`, `__name__`)?

Reply:
How wrap_patch works

https://github.com/mlflow/mlflow/blob/44b152a87e07fce49af3e3a262f7b3caf266bed8/mlflow/utils/autologging_utils.py#L103-L111

Reply:
sure i will take a look into this. In the current behaviour, the patched `fit` method  is responsible for starting and ending the mlflow run using `_run_and_log_function`. if we are going to use wrap_patch how do we pass the arguments for `_run_and_log_function` (since wrap_patch doesnt return anything)?

Reply:
> how do we pass the arguments for _run_and_log_function?

`_run_and_log_function` is called with all the required parameters in `fit`.

Reply:
@harupy Thank you very much. i have updated the code to use `wrap_patch`